dojo/unittests/tools/test_intsights_parser.py

```diff
@@ -0,0 +1,53 @@
+from django.test import TestCase
+from dojo.tools.intsights.parser import IntSightsParser
+from dojo.models import Engagement, Product, Test
+
+
+class TestIntSightsParser(TestCase):
+    def get_test(self):
+        test = Test()
+        test.engagement = Engagement()
+        test.engagement.product = Product()
+        return test
+
+    def test_intsights_parser_without_file_has_no_findings(self):
+        parser = IntSightsParser()
+        findings = parser.get_findings(None, self.get_test())
+        self.assertEqual(0, len(findings))
+
+    def test_intsights_parser_with_no_vuln_has_no_findings(self):
+        testfile = open("dojo/unittests/scans/intsights/intsights_zero_vul.json")
+        parser = IntSightsParser()
+        findings = parser.get_findings(testfile, self.get_test())
+        testfile.close()
+        self.assertEqual(0, len(findings))
+
+    def test_intsights_parser_with_one_critical_vuln_has_one_findings(self):
+        testfile = open("dojo/unittests/scans/intsights/intsights_one_vul.json")
+        parser = IntSightsParser()
+        findings = parser.get_findings(testfile, self.get_test())
+        testfile.close()
+        self.assertEqual(1, len(findings))
+
+        finding = list(findings)[0]
+
+        self.assertEqual("5c80dbf83b4a3900078b6be6", finding.unique_id_from_tool)
+        self.assertEqual("HTTP headers weakness in initech.com web server", finding.title)
+
+    def test_intsights_parser_with_many_vuln_has_many_findings(self):
+        testfile = open("dojo/unittests/scans/intsights/intsights_many_vul.json")
+        parser = IntSightsParser()
+        findings = parser.get_findings(testfile, self.get_test())
+        testfile.close()
+        self.assertEqual(3, len(findings))
+
```

Comment:
Could you add more checks here?
I saw an attribute called "FoundDate": "2019-01-21T15:57:25.709Z" in reports, could be a good candidate for `finding.date`

Reply:
The FoundDate is when the threat intel service found the issue. I wanted to keep that separate from when it was entered into Dojo as an issue.

How are other issues/findings dealing with this kind of issue?

Reply:
You're right if it's the "found" date in the external DB. I was guessing it's was when the finding was detected.
Other parsers use the data of the finding or the date of the report (all finding will have the same date).
You can also take a look at other parser recently updated to see how the date is set (attribute `date`) and checked in the unit tests.